holoviews/plotting/bokeh/element.py

```diff
@@ -178,8 +178,14 @@ def _get_hover_data(self, data, element, empty=False):
 
 
     def _axes_props(self, plots, subplots, element, ranges):
+        # Get the bottom layer and range element
         el = element.traverse(lambda x: x, [Element])
         el = el[0] if el else element
+        if self.batched and not isinstance(self, OverlayPlot):
+            range_el = el
+        else:
+            range_el = element
+
```

Comment:
1. This could be defined lower down closer to where ``range_el`` is used.
2. As ``range_el`` is only used once, I would consider inlining it:

```python
l, b, r, t = self.get_extents(el if self.batched 
        and not isinstance(self, OverlayPlot) else element, ranges)